launcher/modplatform/modrinth/ModrinthPackIndex.cpp

```diff
@@ -55,7 +55,8 @@ void Modrinth::loadIndexedPackVersions(Modrinth::IndexedPack & pack, QJsonArray
         // Find correct file (needed in cases where one version may have multiple files)
         // Will default to the last one if there's no primary (though I think Modrinth requires that
         // at least one file is primary, idk)
-        while (i < files.count()){
+        // NOTE: files.count() is 1-indexed, so we need to subtract 1 to become 0-indexed
```

Comment:
It's not that the count is 1-indexed, it's just that it's the actual count. So if one element exists, count is one but it's on index 0.